### PR TITLE
use less _element_constructor

### PR DESCRIPTION
**Pull request by fchapoton**
trying to avoid ` _element_constructor` with no final underscore

the only serious change is the one in finite fields morphisms.

### :memo: Checklist

- [x] The title is concise, informative, and self-explanatory.
- [x] The description explains in detail what this PR is about.

### CODE REVIEW

File: src/sage/combinat/posets/hasse_cython.pyx [hunk 1]
```diff
@@ -71,7 +71,7 @@ class IncreasingChains(RecursivelyEnumeratedSet_forest):
             self._greater_than = positions
             self._vertices = list(range(n))
 
-        self._element_constructor = element_constructor
+        self._constructor_ = element_constructor
```
Comment (mkoeppe) [inline src/sage/combinat/posets/hasse_cython.pyx]: This name `_constructor_` so far does not appear anywhere in our codebase. Why is this an improvement?

Comment (fchapoton) [reply src/sage/combinat/posets/hasse_cython.pyx]: because it is now a specific name used only in this context, not related to any general mechanism

Comment (mkoeppe) [reply src/sage/combinat/posets/hasse_cython.pyx]: But why use `_constructor_`, not `_constructor`?

Comment (fchapoton) [reply src/sage/combinat/posets/hasse_cython.pyx]: better now ?

File: src/sage/combinat/posets/hasse_cython.pyx [hunk 2]
```diff
@@ -30,8 +30,8 @@ class IncreasingChains(RecursivelyEnumeratedSet_forest):
     - ``positions`` -- a list of sets of integers describing the poset,
       as given by the lazy attribute ``_leq_storage`` of Hasse diagrams.
 
-    - ``element_constructor`` -- used to determine the type of chains,
-      for example ``list`` or ``tuple``
+    - ``constructor`` -- used to determine the type of chains,
+      for example :class:`list` or :class:`tuple`
```
Comment (mkoeppe) [inline src/sage/combinat/posets/hasse_cython.pyx:34]: Can this be renamed without deprecation for the old name?

Comment (fchapoton) [reply src/sage/combinat/posets/hasse_cython.pyx:34]: this is not a named argument, just an argument

Comment (mkoeppe) [reply src/sage/combinat/posets/hasse_cython.pyx:34]: All arguments are named arguments unless you use `, /,`. Users can do `D = IncreasingChains([{0,1},{1}], element_constructor=list, exclude=[]); D`

Comment (fchapoton) [reply src/sage/combinat/posets/hasse_cython.pyx:34]: We have `rename_keyword` but only for keywords, not for args. If you insist, I can make an ad-hoc warning, but I would say it's ok not to deprecate. In particular, it's not in the global namespace.

Comment (mkoeppe) [reply src/sage/combinat/posets/hasse_cython.pyx:34]: maybe the name of the parameter can just be left as is?

Comment (fchapoton) [reply src/sage/combinat/posets/hasse_cython.pyx:34]: ok, done

**Review by mkoeppe — APPROVED** [review]
LGTM, thanks.

**Pull request merged**